### ISSUE TITLE
Eigen update needed to fix tensorflow build failure on ppc64le

### ISSUE THREAD

**Issue by npanpaliya** [tensorflow/tensorflow#48664]
Request to update Eigen in TF to https://gitlab.com/libeigen/eigen/-/commit/06c2760bd1139711eeffa30266ead43423891698
to fix compilation issue on ppc64le.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux CentOS 7
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source master branch
- TensorFlow version:
- Python version: 3.9
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source): 3.7.2
- GCC/Compiler version (if compiling from source): gcc 9, gcc8, gcc 7
- CUDA/cuDNN version: NO CUDA (CPU only)
- GPU model and memory:

**Describe the problem**
While building tensorflow on ibmcom/tensorflow-ppc64le:devel-manylinux2014 docker image for ppc64le and using gcc 9 (tried gcc 8 and gcc 7 too), we are seeing a compilation error related to Eigen. The fix for the same has been pushed to Eigen repository yesterday at https://gitlab.com/libeigen/eigen/-/commit/06c2760bd1139711eeffa30266ead43423891698
If TF updates the eigen version to the above mentioned commit, we will get the issue fixed without any patching.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

```
BAZEL_LINKLIBS=-l%:libstdc++.a bazel build -c opt --config=v2 --local_cpu_resources 4 --local_ram_resources 4096 \
    //tensorflow/tools/pip_package:build_pip_package
bazel-bin/tensorflow/tools/pip_package/build_pip_package --nightly_flag --cpu $WORKSPACE/tensorflow_pkg```

```

Error:

```
tensorflow/compiler/xla/service/cpu/runtime_single_threaded_matmul.cc:121:76:   required from here
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:498:20: error: taking address of rvalue [-fpermissive]
  498 |             lhs0 = &lhs(j + 0, i);
      |                    ^~~~~~~~~~~~~~
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:499:20: error: taking address of rvalue [-fpermissive]
  499 |             lhs1 = &lhs(j + 1, i);
      |                    ^~~~~~~~~~~~~~
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:501:20: error: taking address of rvalue [-fpermissive]
  501 |             lhs0 = &lhs(j + 2, i);
      |                    ^~~~~~~~~~~~~~
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:502:20: error: taking address of rvalue [-fpermissive]
  502 |             lhs1 = &lhs(j + 3, i);
      |                    ^~~~~~~~~~~~~~
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:505:20: error: taking address of rvalue [-fpermissive]
  505 |             lhs0 = &lhs(i, j + 0);
      |                    ^~~~~~~~~~~~~~
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:506:20: error: taking address of rvalue [-fpermissive]
  506 |             lhs1 = &lhs(i, j + 1);
      |                    ^~~~~~~~~~~~~~
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:508:20: error: taking address of rvalue [-fpermissive]
  508 |             lhs0 = &lhs(i, j + 2);
      |                    ^~~~~~~~~~~~~~
external/eigen_archive/Eigen/src/Core/arch/AltiVec/MatrixProduct.h:509:20: error: taking address of rvalue [-fpermissive]
  509 |             lhs1 = &lhs(i, j + 3);
      |                    ^~~~~~~~~~~~~~
```

**Comment by mihaimaruseac** [comment]
I think this should be resolved now?

**Comment by gadagashwini** [comment]
@npanpaliya,
Looks like issue is resolved, Take a look at similar thread [#45746](https://github.com/tensorflow/tensorflow/issues/45746#issuecomment-997969395). Thanks!
